compiler/src/iree/compiler/Dialect/Stream/Transforms/ReplicateGlobalsPerAffinity.cpp

```diff
@@ -516,6 +522,16 @@ void ReplicateGlobalsPerAffinityPass::runOnOperation() {
 
     IREE::Stream::AffinityAttr executionAffinityAttr = *affinityAttrs.begin();
     LDBG() << "updating operands with the affinity: " << executionAffinityAttr;
+
+    // Order matters here because it is wrong if an operand is already updated
+    // with a new value. E.g.,
+    //   %0 = flow.dispatch @dispatch0(%global)
+    //   %1 = flow.reshape %0 ...
+    // If we update %0 first, then when updating %1 we will see a new global
+    // from %0 which uses the new global, that is not cached in
+    // `globalPerAffinityHelper`.
+    // Thus, we collect all the updates first, then apply them.
+    SmallVector<std::pair<OpOperand *, Value>> updateList;
```

Comment:
Is there a specific reason to use `std::pair` here? I noticed that in other places in this file, `std::tuple` is used, even with just two members.